amip_interp/amip_interp.F90

```diff
@@ -265,31 +274,48 @@ module amip_interp_mod
 !!
 !> @ingroup amip_interp_mod
 interface amip_interp_new
-   module procedure amip_interp_new_1d
-   module procedure amip_interp_new_2d
+   module procedure amip_interp_new_1d_r4, amip_interp_new_1d_r8
+   module procedure amip_interp_new_2d_r4, amip_interp_new_2d_r8
 end interface
 
+!> Initialize amip_interp. To explicitly initialize the module in r4 mode or r8 mode,
+!! call either amip_interp_init_r4 or amip_interp_init_r8. amip_interp_init is an
+!! alias for amip_interp_init_r8.
+interface amip_interp_init
+  module procedure amip_interp_init_r8
+end interface
+
+!> Frees data associated with a amip_interp_type variable. Should be used for any
+!! variables initialized via @ref amip_interp_new.
+!> @param[inout] Interp A defined data type variable initialized by amip_interp_new and used
+!! when calling get_amip_sst and get_amip_ice.
+interface amip_interp_del
+  module procedure amip_interp_del_r4, amip_interp_del_r8
+end interface amip_interp_del
 
 !----- public data type ------
 
-!> @brief Contains information needed by the interpolation module (exchange_mod) and buffers data.
+!> @brief Contains information needed by the interpolation module (exchange_mod) and buffers
+!! data. Real precision is determined by the rkind parameter, which defaults to r8_kind.
 !> @ingroup amip_interp_mod
-type amip_interp_type
+type amip_interp_type(rkind)
+   integer, kind            :: rkind = r8_kind ! Real kind for data1 and data2
```

Comment:
I think we want to avoid using parameterized derived types for now

Reply:
I've replaced `amip_interp_type` with `amip_interp_type_r4` and `amip_interp_type_r8` in 323ef7b7da84b0ae73e535c9adf2a20e86a332ee. I don't think there's a way to export the name `amip_interp_type` as an alias for `amip_interp_type_r8`, so minor changes to the model code will be required. Fortunately, I believe `ice_amip` and `ocean_amip` are the only model codes which use this module.